routes/signup.js

```diff
@@ -31,85 +32,226 @@ var connection = mysql.createConnection({
-      // Redirect to next page (first user page).
-      res.send({
-        "code":200,
-        "success":"user registered sucessfully"
+
+
+router.post('/register', function(req, res) {
+  console.log(req.body);
+
+//First we make a query to see if user exists in the database
+  connection.query(
+    'SELECT * FROM users WHERE email = ?',
+    [req.body.email],
+    function(error, result, fields) {
+      if (error) throw error;
+
+//if we find the user exists in the database, we send "User already exists" to the client
+      // if (result) return res.status(404).json({ error: 'User already exists' });
+      console.log("successfully query", result);
+
+      bcrypt.genSalt(10, function(err, salt) {
+
+        bcrypt.hash(req.body.password, salt, function(err, password_hash) {
+
+//1) insert into users table
+          connection.query('INSERT INTO users (email, password, username, email_verification_token) VALUES (?, ?, ?, ?)',
+          [req.body.email, password_hash, req.body.username, uuidv4()],
+          function (error, results, fields) {
+            if (error) {
+              console.log(error)
+            } else {
+              //send a notificiation to the client side to let user verify their email
+              res.json({
+                  message: "We sent you an email for email verification. Please confirm your email."
+              });
+
+              //query the new inserted user to get the user-id and email verification code
+              connection.query(
+                'SELECT * FROM users WHERE email = ?',
+                [req.body.email],
+                function(error, result, fields) {
+                  if (error) throw error;
+                   //use sendgrid to send email
+                  const email_verification = {
+                    to: req.body.email,
+                    from: 'simon@acceptmycrypto.com',
+                    subject: 'Please click the link below to verify your email!',
+                    text: 'Thank You for signing up! Go to this url to complete the registration.',
+                    html: `<a href="http://localhost:3001/email-verify/${result.id}/${result.email_verification_token}>Verify My Email</a>`
+                  };
+                  sgMail.send(email_verification);
+                }
+              );
+
+              //Once the user clicks on the email verification, we get the id and email verification params
+              router.get('/email-verify/:user_id/:email_verification_token', function(req, res) {
+                console.log("params: ", req.params);
+                connection.query(
+                  'SELECT * FROM users WHERE id = ?',
+                  [req.params.user_id],
+                  function(error, result, fields) {
+                    if (error) throw error;
+
+                    //if user is verified already then send a message to user that the account is verified
+                    if (result.verified_email === 0) {
```

Comment:
```suggestion
                    if (result.verified_email === 1) {
```
Are you checking on line 95 if the user has previously verified their email? In that case verified_email would be `1` which is true.

Reply:
ahhh yeah you're right haha I got that part messed up